### PR TITLE
feat(safe-apps): Add custom app redesign

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves https://github.com/safe-global/web-core/issues/843

## How this PR fixes it
We are updating the design for de "Add Custom app modal" feature and make the UI/UX better

[New Design](https://www.figma.com/file/rEKs2w2zMdmhNtkgTQDEWD/Add-custom-app?node-id=2%3A2757)

### CODE REVIEW

File: src/components/safe-apps/AddCustomAppModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  }, [chainId, debouncedUrl])
+
+  const handleClose = () => {
+    reset()
+    onClose()
+  }
+
+  const isAppAlreadyInTheList = useCallback(
+    (appUrl: string) => safeAppsList.some((app) => isSameUrl(app.url, appUrl)),
+    [],
```
Comment (mikhailxyz) [inline src/components/safe-apps/AddCustomAppModal/index.tsx:90]: missing `safeAppsList` dependency

File: src/components/safe-apps/AddCustomAppModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  appUrl: string
+  riskAcknowledgement: boolean
+  safeApp: SafeAppData
+}
+
+const HELP_LINK = 'https://docs.gnosis-safe.io/build/sdks/safe-apps'
+
+const AddCustomAppModal = ({ open, onClose, onSave, safeAppsList }: Props) => {
+  const chainId = useChainId()
+  const currentChain = useCurrentChain()
```
Comment (mikhailxyz) [inline src/components/safe-apps/AddCustomAppModal/index.tsx:52]: does `currentChain` contain `chainId`? if yes, you can get it from there instead calling `useChainId`

File: src/components/safe-apps/AddCustomAppModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  const isAppAlreadyInTheList = useCallback(
+    (appUrl: string) => safeAppsList.some((app) => isSameUrl(app.url, appUrl)),
+    [],
+  )
+
+  const shareUrl = `${window.location.origin}${AppRoutes.share.safeApp}?appUrl=${encodeURIComponent(
+    safeApp?.url || '',
+  )}&chain=${currentChain?.shortName}`
+  const isSafeAppValid = isValid && safeApp
+  const isCustomAppInTheDefaultList = errors?.appUrl?.type === 'doesntExist'
```
Comment (mikhailxyz) [inline src/components/safe-apps/AddCustomAppModal/index.tsx:97]: should it be `alreadyExists` instead of `doesntExist`?

**Review by mikhailxyz — APPROVED** [review]

File: src/services/safe-apps/manifest.ts [hunk 4]
```diff
@@ -52,12 +52,7 @@ const getAppLogoUrl = (appUrl: string, { icons = [], iconPath = '' }: AppManifes
     return iconUrl
   }
 
-  if (isRelativeUrl(iconUrl)) {
-    const appUrlHost = new URL(appUrl).host
-    return `${appUrlHost}${iconUrl}`
-  }
-
-  return `${appUrl}/${iconUrl}`
+  return `${appUrl}${isRelativeUrl(iconUrl) ? '' : '/'}${iconUrl}`
```
Comment (mikhailxyz) [inline src/services/safe-apps/manifest.ts:55]: what happened here? how does this work when the URL is not relative?

**Comment by JagoFigueroa** [comment]
Looking and working like a charm, thanks!

**Pull request merged**